core/blockchain.go

```diff
@@ -2183,10 +2183,27 @@ func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 	}
 	// Both sides of the reorg are at the same number, reduce both until the common
 	// ancestor is found
+
+	log.Info("[DEBUG] About to perform reorg", "old block", oldBlock.NumberU64(), "old block hash", oldBlock.Hash(), "new block", newBlock.NumberU64(), "new block hash", newBlock.Hash())
+
 	for {
 		// If the common ancestor was found, bail out
 		if oldBlock.Hash() == newBlock.Hash() {
 			commonBlock = oldBlock
+
+			log.Info("[DEBUG] Found common ancestor in reorg", "number", commonBlock.NumberU64())
+			if len(oldChain) > 0 {
+				log.Info("[DEBUG] Old chain", "start", oldChain[0], "end", oldChain[len(oldChain)-1], "length", len(oldChain))
+			} else {
+				log.Info("[DEBUG] Empty old chain")
+			}
+
+			if len(newChain) > 0 {
+				log.Info("[DEBUG] New chain", "start", newChain[0], "end", newChain[len(newChain)-1], "length", len(newChain))
+			} else {
+				log.Info("[DEBUG] Empty new chain")
```

Comment:
Maybe it's also insightful to log the total difficulty and the respective authors as well ?

Reply:
Sure added the difficulty for now. For author the block header don't have that data so skipping that for now.